sf-stream/src/runtime.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::metrics;
+
+use aptos_api::context::Context;
+use aptos_api_types::TransactionOnChainData;
+use aptos_config::config::NodeConfig;
+use aptos_logger::{debug, error, warn};
+use aptos_types::chain_id::ChainId;
+use futures::channel::mpsc::channel;
+use std::time::Duration;
+use std::{net::SocketAddr, sync::Arc};
+use storage_interface::DbReader;
+use tokio::runtime::{Builder, Runtime};
+use tokio::time::sleep;
+
+/// Creates a runtime which creates a thread pool which pushes firehose of block protobuf to SF endpoint
+/// Returns corresponding Tokio runtime
+pub fn bootstrap(
+    config: &NodeConfig,
+    chain_id: ChainId,
+    db: Arc<dyn DbReader>,
+) -> anyhow::Result<Runtime> {
+    let runtime = Builder::new_multi_thread()
+        .thread_name("sf-stream")
+        .enable_all()
+        .build()
+        .expect("[sf-stream] failed to create runtime");
+
+    let node_config = config.clone();
+
+    // TODO: put this into an arg param
+    let starting_version: Option<u64> = None;
```

Comment:
We should also infer this if it's not explicitly specified right? The user will probably specify the start version the very first time they start the node but in subsequent reboots we'd want to continue from last known version I'd imagine.

Reply:
This part is stateless; `put this into an arg param` - will create a `--sf-stream-starting-version` or `*-block`, which is what we'd use for the stateful part :-)